rpcs3/Emu/RSX/RSXThread.h

```diff
@@ -879,7 +879,7 @@ namespace rsx
 
 		reservation_lock(u32 addr, u32 length, bool setting)
 		{
-			if (setting && addr < constants::local_mem_base)
+			if (setting)
```

Comment:
Don't apply reservation lock for rawspu mmio or RSX local mem. We don't need to keep adding address checks all over and this covers most of that.

Reply:
There are no CELL reservations in MMIO or local RSX memory, it would crash the PS3 program. This only affects DMA as intended.

Reply:
If anything we also need to add a check for zero pages here. Nothing good lives in high memory or zero page. It's either RSX mem, stack, SPU mmio or invalid addresses.

Reply:
> There are no CELL reservations in MMIO or local RSX memory, it would crash the PS3 program. This only affects DMA as intended. (accounts for local memory)

This check avoids doing complex work unnecessarily and is a single instruction. Remember we invoke checks without branching in RSX code as well for this. Instead of having checks littered all over the place I chose to have the callee discard invalid requests. It is just cleaner that way.

Reply:
I need SRM to account for local rsx memory, but FIFO doesn't need it.

Reply:
The problem is that we have rsx reservation locks in RSX code where this is undesired.

Reply:
But this constructor overload is SPU and RSX FIFO  only.

Reply:
Actually nvm, let's keep it. I'll revise the locking mechanism later.